### PR TITLE
tracers: rework chain trace

### PR DESCRIPTION
**Pull request by yoomee1313**
## Proposed changes

- This PR is brought from ethereum to fix the oom issue when calling debug api repeatedly. (e.g. once every 10 seconds)

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [ ] New feature or enhancement
- [ ] Others

## Checklist

_Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have read the [CONTRIBUTING GUIDELINES](https://github.com/kaiachain/kaia/blob/main/CONTRIBUTING.md) doc
- [ ] I have read the [CLA](https://gist.github.com/kaiachain-dev/bbf65cc330275c057463c4c94ce787a6) and signed by comment `I have read the CLA Document and I hereby sign the CLA` in first time contribute
- [ ] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- https://github.com/kaiachain/kaia/issues/75

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: node/cn/tracers/api_test.go [hunk 1]
```diff
@@ -171,17 +182,74 @@ func (b *testBackend) StateAtTransaction(ctx context.Context, block *types.Block
...
-			return nil, vm.BlockContext{}, vm.TxContext{}, nil, fmt.Errorf("transaction %#x failed: %v", tx.Hash(), err)
+			return nil, vm.BlockContext{}, vm.TxContext{}, nil, nil, fmt.Errorf("transaction %#x failed: %v", tx.Hash(), err)
 		}
 		statedb.Finalise(true, true)
 	}
-	return nil, vm.BlockContext{}, vm.TxContext{}, nil, fmt.Errorf("transaction index %d out of range for block %#x", txIndex, block.Hash())
+	return nil, vm.BlockContext{}, vm.TxContext{}, nil, nil, fmt.Errorf("transaction index %d out of range for block %#x", txIndex, block.Hash())
 }
 
+// TODO-kaia: enable testcode
```
Comment (ian0371) [inline node/cn/tracers/api_test.go:196]: what's this?

File: node/cn/tracers/api.go [hunk 2]
```diff
@@ -297,15 +329,17 @@ func (api *CommonAPI) traceChain(start, end *types.Block, config *TraceConfig, n
 		tasks    = make(chan *blockTraceTask, threads)
 		results  = make(chan *blockTraceTask, threads)
 		localctx = context.Background()
+		reler    = new(releaser)
```
Comment (ian0371) [inline node/cn/tracers/api.go:332]:
```suggestion
		releaser    = new(releaser)
```

Comment (yoomee1313) [reply node/cn/tracers/api.go:332]: The name shadows the struct `releaser`. I think original name is quite okay.

File: node/cn/tracers/api_test.go (hunk 1 above)
Comment (yoomee1313) [reply node/cn/tracers/api_test.go:196]: Oh thank you, I forgot to enable this testcode. Let me enable this.

**Review by blukat29 — APPROVED** [review]

**Review by ian0371 — APPROVED** [review]

**Pull request merged**